modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.downloadsButtonPlugin/mh_downloads.js

```diff
@@ -0,0 +1,272 @@
+              match_tags = true;
+              break;
+            }
+          }
+          // skip by flavor
+          var match_flavors = false;
+          for (var flavorsIndex = 0; flavorsIndex < self._flavors.length; flavorsIndex++) {
+            var flavor = self._flavors[flavorsIndex];
+            if (track.type === flavor) {
+              match_flavors = true;
+              break;
+            }
+          }
+
+          if (self._tags.length > 0 && self._flavors.length > 0) {
+            if (self._tags_and_flavors) {
+              if (!(match_tags && match_flavors)) { continue; }
+            } else {
+              if (!(match_tags || match_flavors)) { continue; }
+            }
+          } else {
+            if (self._tags.length > 0 && !match_tags) { continue; }
+            if (self._flavors.length > 0 && !match_flavors) { continue; }
+          }
+
+          var parsedTrack = new DownloadTrack(track);
+          parsedTracks.push(parsedTrack);
+        }
+      }
+      self._tracks = parsedTracks;
+    }
+
+    renderTracks(domElement) {
+      var self = this;
+      let container = document.createElement('div');
+      container.className = 'downloadsButtonContainer';
+
+      if (self._episode === undefined || self._tracks.length < 1) {
+        container.innerHTML = '<div class="downloadsItemContainer"><p class="noTracks">'
+                                + paella.dictionary.translate('No downloads available') + '</p></div';
+      } else {
+        let trackTypes = ['video', 'audio'];
+        let flavors = self._tracks.reduce(function(result, track) {
+          if (result.indexOf(track.flavor) < 0) {
+            result.push(track.flavor);
+          }
+          return result;
+        }, []);
+
+        // group by flavor
+        for (let flavorsIndex = 0; flavorsIndex < flavors.length; flavorsIndex++) {
+          let flavor = flavors[flavorsIndex];
+          let flavorContainer = document.createElement('div');
+          flavorContainer.className = 'downloadsButtonItemContainer';
+
+          let flavorNameContainer = document.createElement('p');
+          flavorNameContainer.innerText = flavor;
+          flavorContainer.appendChild(flavorNameContainer);
+
+          // order by type (audio or video)
+          for (let trackTypesIndex = 0; trackTypesIndex < trackTypes.length; trackTypesIndex++) {
+            let trackType = trackTypes[trackTypesIndex];
+            let tracks = self._tracks.filter(function(track) {
+              return this === track.type;
+            }, trackType);
+
+            // order by mimetype
+            let trackMimetypes = tracks.reduce(function(result, track) {
+              if (result.indexOf(track.mimetype) < 0) {
+                result.push(track.mimetype);
+              }
+              return result;
+            }, []);
+
+            trackMimetypes.sort(function(f1, f2) {
+              return f1 === f2 ? 0 : f1 > f2 ? 1 : -1;
+            });
```

Comment:
This is just natural sorting, isn't it? So this should be identical to just calling `sort()` without custom function

```suggestion
            trackMimetypes.sort();
```